Enable Linux builds

Nicky-D opened the pull request:
@nat-goodspeed another one for you

Nicky-D committed:
Enable Linux64 builds

Nicky-D committed:
Change library name to libmeshoptimizer.a

Nicky-D committed:
Add missing popd

Nicky-D committed:
Add VERSION.txt generation

nat-goodspeed commented on build-cmd.sh:
```diff
@@ -116,15 +116,26 @@ pushd "$MESHOPT_SOURCE_DIR"
             make -j $AUTOBUILD_CPU_COUNT
             make install
 
+            popd
```
This doesn't cause trouble with the unqualified reference to `LICENSE.md` at line 142, or the later  `popd` at line 143?

Apparently the only relative reference this new `popd` would fix would be `$VERSION_HEADER_FILE` at line 133. Actually, now you've got me curious how that works on [Mac](https://github.com/secondlife/3p-meshoptimizer/actions/runs/7265243028/job/19794507116#step:2:466) or [Windows](https://github.com/secondlife/3p-meshoptimizer/actions/runs/7265243028/job/19794506849#step:2:814), neither of which have a corresponding early `popd`.

Given the `pushd` at line 42, maybe a simpler fix would be to exclude `$MESHOPT_SOURCE_DIR` from `$VERSION_HEADER_FILE`, assuming that doesn't break the Windows or Mac builds?

Nicky-D replied:
It fixes the unqualified `LICENSE.md` reference in line 142

The culprit here is a pushd in line 112. This was missing a corresponding popd before.

While in the Windows and Mac case there is no such thing as an extra pushd for whatever reason for Linux there is one. (I'm a fan of building in a `build` subdirectory, so I rather kept that)

This also explains why Windows/Mac use `cmake .` while Linue uses `cmake ..`

Event though prefixing LICENSE.md would put a patch on that, there would still be no pushd/popd symmetry

nat-goodspeed replied:
Oh my gosh! That `pushd` on line 112 is a bit subtle. I completely overlooked it, which is why I questioned your new corresponding `popd`. Could I ask you to break that out on a separate line as part of this PR?

nat-goodspeed replied:
Realizing line 114 wasn't originally yours, is there a reason to keep the trailing backslash or empty line 115?

Nicky-D committed:
Split pushd into it's own line and indent statements between pushd/popd

Having the pushd tacked at the end of the mkdir statement can be confusing and easy to overlook.
-> Break the statement into two lines and indent the code

Nicky-D committed:
Remove trailing backslash

nat-goodspeed commented:
https://github.com/secondlife/3p-meshoptimizer/releases/tag/v160-4f905dd
